internal/controllers/cluster/cluster_controller_status_test.go

```diff
@@ -907,16 +992,174 @@ func TestSetMachinesReadyCondition(t *testing.T) {
 			if tt.machines != nil {
 				machines = collections.FromMachines(tt.machines...)
 			}
-			setMachinesReadyCondition(ctx, tt.cluster, machines, tt.getDescendantsSucceeded)
+			setWorkerMachinesReadyCondition(ctx, tt.cluster, machines, tt.getDescendantsSucceeded)
 
-			condition := v1beta2conditions.Get(tt.cluster, clusterv1.ClusterMachinesReadyV1Beta2Condition)
+			condition := v1beta2conditions.Get(tt.cluster, clusterv1.ClusterWorkerMachinesReadyV1Beta2Condition)
 			g.Expect(condition).ToNot(BeNil())
 			g.Expect(*condition).To(v1beta2conditions.MatchCondition(tt.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
 		})
 	}
 }
 
-func TestSetMachinesUpToDateCondition(t *testing.T) {
+func TestSetControlPlaneMachinesUpToDateCondition(t *testing.T) {
+	tests := []struct {
+		name                    string
+		cluster                 *clusterv1.Cluster
+		machines                []*clusterv1.Machine
+		getDescendantsSucceeded bool
+		expectCondition         metav1.Condition
+	}{
+		{
+			name:                    "get descendant failed",
+			cluster:                 fakeCluster("c"),
+			machines:                nil,
+			getDescendantsSucceeded: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.ClusterControlPlaneMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.ClusterControlPlaneMachinesUpToDateInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:                    "no machines",
+			cluster:                 fakeCluster("c"),
+			machines:                []*clusterv1.Machine{},
+			getDescendantsSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.ClusterControlPlaneMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.ClusterControlPlaneMachinesUpToDateNoReplicasV1Beta2Reason,
+				Message: "",
+			},
+		},
+		{
+			name:    "One machine up-to-date",
+			cluster: fakeCluster("c"),
+			machines: []*clusterv1.Machine{
+				fakeMachine("up-to-date-1", v1beta2Condition(metav1.Condition{
```

Comment:
I know it doesn't change the test result, but what about adding `controlPlane(true)`, for machines in this test like you did for TestSetControlPlaneMachinesReadyCondition